Better controls option menu

Oganesson897 opened the pull request:
It looks like Controlling mod because I ported by that.

ZZZank commented:
Please provide more details about this PR, e.g. API changes, what have been tested, screenshots.

And honestly, I dont think this is necessary. Controlling is doing pretty good in terms of Cleanroom compatibility, integrating it directly (without adding features) is not any better than installing an individual mod.

Ecdcaeb commented on src/main/java/us/getfluxed/controlsearch/client/gui/EnumSortingType.java:
```diff
@@ -0,0 +1,24 @@
+
+public enum EnumSortingType {
+
+    DEFAULT, AZ, ZA;
```
I noticed that there have had an sorting type here for forge.
https://github.com/CleanroomMC/Cleanroom/blob/main/src/main/java/net/minecraftforge/fml/client/GuiModList.java#L75

Oganesson897 replied:
fix later soooooory

Oganesson897 commented:
Okay, so the port reason is [this project item](https://github.com/orgs/CleanroomMC/projects/4?pane=issue&itemId=25411095).
I hope I'm not misunderstanding.
the first sentence is `includes better UI` so I made this port.

About the API changes:
I violently copied the `Controlling` code straight over.
Unlike installing `Controlling` directly, I modified the vanilla GUI directly in Forge Patches.
This means that if any other mod modifies the `GuiControls` class or the `GuiKeyBindingList` class using `Mixin` or the like, there will be predictable problems.
And I also directly changed the type of one of the member variables, and while it was private protected and I fixed most of the calls associated with it, there's no guarantee that other mods will call that variable elsewhere.
I will be fix this problem later.

About tested:
It can work like origin `Controlling`, and I add some language support (btw it is less than original).
I am sorry I am not working on my PC now so I will put some screenshots later.

And honestly, I dont think include Controlling is a good idea but I think we need a better controls menu.
